src/Domain/Service/Precondition/README.md

```diff
@@ -10,6 +10,17 @@ This diagram depicts the detailed interfaces and relationships. New concrete pre
 
 <div align="center"><img src="resources/detail.png" alt="Detail diagram" /></div>
 
-Below is a depiction of the actual hierarchy of preconditions. In short, all domain operations have the same preconditions except the beginner, which has a different relationship with the staging directory.
+Below is a depiction of the actual hierarchy of preconditions.
 
 <div align="center"><img src="resources/hierarchy.png" alt="Hierarchy diagram" /></div>
+
+## Symlinks
+
+The reasoning behind the preconditions related to symlinks and hard links is not necessarily obvious and bears special explanation:
+
+- **No absolute links** ([`NoAbsoluteLinksExist`](NoAbsoluteLinksExistInterface.php)) - An absolute link in the active directory will point to the same path on disk when copied to the staging directory, creating the possibility of staged operations changing production files.
+- **No hard links** ([`NoHardLinksExist`](NoHardLinksExistInterface.php)) - Hard links point to inodes as opposed to mere paths, i.e., the actual data blocks on the disk. Therefore, a hard link _to_ a file essentially _is_ that file. So anything done to one is done to the other, with the same net effect as absolute links.
+- **No soft links that point outside the codebase** ([`NoLinksPointOutsideTheCodebase`](NoLinksPointOutsideTheCodebaseInterface.php)) - Active and staging directories adjacent to one another on the filesystem have identical ancestors by definition. Hence, a relative link pointing out of one would target the same file as the corresponding link "next door" and, like previous scenarios, create the possibility of corrupting accidentally shared files. (Links that point _within_ the codebase pose no threat--even if they cross internal package boundaries.)
```

Comment:
Thoughts? 😊

Reply:
Added:

> ...even if they cross internal package boundaries, i.e., Composer or other vendor libraries. The greatest risk in that case is probably links with missing targets, which seems improbable with low impact--it would represent an awfully bizarre bug on the part of the offending package and would pose no known danger to any Composer Stager operations.